datafusion/physical-expr/src/expressions/cast.rs

```diff
@@ -76,6 +76,26 @@ impl CastExpr {
     pub fn cast_options(&self) -> &CastOptions<'static> {
         &self.cast_options
     }
+    pub fn is_bigger_cast(&self, src: DataType) -> bool {
```

Comment:
I think this is similar to what is in the coercion logic. Maybe we can consolidate it at some point

Reply:
I can imagine this util would be useful in another context.